### DISCUSSION TITLE
umi启动提示内存泄露问题？

### DISCUSSION THREAD

**Discussion by joraboy328** [umijs/umi#11116] [General]
UmiJS3 启动告警如下：
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.at Overflow..........

查询网上说：这是由于在组件已被卸载后仍然尝试修改组件的状态所导致的问题。在组件被卸载后，无法使用setState来改变组件的状态，这会导致React引擎出现内存泄漏警告。

我发现配置路由菜单时，如果一旦设定name属性，就报错，如何解决此问题
routes:
[{ 
    path: '/login', 
    component: '@/pages/login/index',
    name:'登录',
    layout:false,
    hideInMenu:true
},]

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4

**Reply by joraboy328** [reply]
按照官网升级到umi 4，也同样出现此问题？请教如何解决呢？多谢！

**Reply by fz6m** [reply]
给一个会发生此问题的最小复现看看。

**Reply by joraboy328** [reply]
在设定路由/菜单配置时候，name属性设定会造成此问题，奇怪

routes：[
{ 
    //name:'首页', //### **如果不设置name属性就不出现此报错，一旦设置了就报错**
    path: '/', 
    name: '首页',
    component: '@/pages/index', 
   
},]

**Reply by joraboy328** [reply]
完整配置代码如下：

export default defineConfig({

  nodeModulesTransform: {
    type: 'none',
  },

  antd:{
  },
  dva:{
    immer:false
  },

  layout:{
    name:'xxx',
    contentWidth: 'Fluid',
    splitMenus: false,
    colorWeak:false,
    layout:'mix',
    pure:false,
    locale:true,

  },

  routes: routes：[
   {
       //name:'首页', //### 如果不设置name属性就不出现此报错，一旦设置了就报错！
         path: '/',
         name: '首页',
         component: '@/pages/index',
},]

  fastRefresh: {},
});

**Reply by fz6m** [reply]
需要提供一个最小复现的仓库

**Reply by joraboy328** [reply]
> 需要提供一个最小复现的仓库

https://gitee.com/joraboy328/myumijs3.git
谢谢！

**Reply by fz6m** [reply]
umi 3 不维护了，建议升级到 umi 4 。

**Reply by zhenyu-dev** [reply]
4也有这个问题，但应该是pro-components的问题
